### DISCUSSION TITLE
Supabase function and trigger help

### DISCUSSION THREAD

**Discussion by Othman-Sami** [supabase/supabase#21670] [Questions]
Hi everyone

I want your help to explain function in Supabase 

I have two table ( request , user )
request ( email ,name , city )
user ( id , email , name , city ) 

in table request I have all information about people who want to be add in my app

I make trigger to do function every time when new account add , to do function for have value in user table from request table when row that equal to email ,

I make run and have success 

but when I add new user I had error in database 
and no account create

```
create
or replace function public.handle_new_user () returns trigger as $$
DECLARE

  name text;
  city text;

BEGIN
    -- Retrieve the 'name' value from the 'request' table based on the matching email
    SELECT name , city INTO name , city 
    FROM public.request
    WHERE public.request.email = new.email;
    
    -- Insert the values into the 'public.user' table
    INSERT INTO public.user (id, email,  name , city )
    VALUES (NEW.id, NEW.email,  name ,city );
    
    RETURN NEW;
END;
$$ language plpgsql security definer;

-- Create the trigger to execute the function after insert on auth.users
create trigger on_auth_user_created
after insert on auth.users for each row
execute procedure public.handle_new_user ();
```

**Comment by GaryAustin1** [comment]
Look in the Postgres logs and see what the error detail is.

**Reply by Othman-Sami** [reply]
column "email" of relation "user" does not exist . 

But I have column in request table named "email"

**Reply by GaryAustin1** [reply]
You marked this as closed did that give you the info you needed to find the issue?
